onnxruntime/core/framework/graph_partitioner.cc

```diff
@@ -239,6 +239,26 @@ static Status GetCapabilityForEP(const GetCapabilityForEPParams& params) {
 }
 
 #if !defined(ORT_MINIMAL_BUILD)
+
+// This function queries the capabilities for a given EP, but it does not assign the nodes.
+// It also does not perform layout transformation. This will be done during normal partitioning.
+static Status GetCapabilityForEPForAotInlining(const Graph& graph,
+                                               const KernelRegistryManager& kernel_registry_mgr,
+                                               const IExecutionProvider& current_ep,
+                                               std::vector<std::unique_ptr<ComputeCapability>>& capabilities) {
+  const auto& ep_type = current_ep.Type();
+
+  const auto kernel_registries_for_ep = kernel_registry_mgr.GetKernelRegistriesByProviderType(ep_type);
+  const KernelLookup kernel_lookup{ep_type,
+                                   kernel_registries_for_ep,
+                                   kernel_registry_mgr.GetKernelTypeStrResolver()};
+
+  const GraphViewer graph_viewer(graph);
+  capabilities = get_capabilities(current_ep, graph_viewer, kernel_lookup);
```

Comment:
I think we need to add a TODO here to provide EPs the capability to look inside functions.

Reply:
The inlining takes place layer by layer. At each layer each EP has an ability to look at the inlined function ops and examine them.